integrationTests/chainSimulator/bridge/esdtSafe_test.go

```diff
@@ -0,0 +1,518 @@
+	supply = big.NewInt(234673347)
+	createArgs = createNftArgs(collectionId, supply, "META #1")
+	chainSim.SendTransactionWithSuccess(t, cs, account.Wallet.Bytes, &account.Nonce, account.Wallet.Bytes, chainSim.ZeroValue, createArgs, uint64(60000000))
+	token = chainSim.ArgsDepositToken{
+		Identifier: collectionId,
+		Nonce:      uint64(1),
+		Amount:     supply,
+		Type:       core.MetaFungible,
+	}
+	wallets[&account] = token
+
+	account = createNewAccount(cs, &accountShardId)
+	collectionId = chainSim.RegisterAndSetAllRolesDynamic(t, cs, account.Wallet.Bytes, &account.Nonce, issueCost, "DMETA", "DMETA", core.DynamicMetaESDT, 10)
+	supply = big.NewInt(64865382)
+	createArgs = createNftArgs(collectionId, supply, "DMETA #1")
+	chainSim.SendTransactionWithSuccess(t, cs, account.Wallet.Bytes, &account.Nonce, account.Wallet.Bytes, chainSim.ZeroValue, createArgs, uint64(60000000))
+	token = chainSim.ArgsDepositToken{
+		Identifier: collectionId,
+		Nonce:      uint64(1),
+		Amount:     supply,
+		Type:       core.DynamicMeta,
+	}
+	wallets[&account] = token
+
+	return wallets
+}
+
+func createNewAccount(cs chainSim.ChainSimulator, shardId *uint32) chainSim.Account {
+	wallet, _ := cs.GenerateAndMintWalletAddress(*shardId, chainSim.InitialAmount)
+	nextShardId(shardId)
+	_ = cs.GenerateBlocks(1)
+
+	return chainSim.Account{
+		Wallet: wallet,
+		Nonce:  uint64(0),
+	}
+}
+
+func nextShardId(shardId *uint32) {
+	*shardId++
+	if *shardId > 2 {
+		*shardId = 0
+	}
+}
+
+func createNftArgs(tokenIdentifier string, initialSupply *big.Int, name string) string {
+	return "ESDTNFTCreate" +
+		"@" + hex.EncodeToString([]byte(tokenIdentifier)) +
+		"@" + hex.EncodeToString(initialSupply.Bytes()) +
+		"@" + hex.EncodeToString([]byte(name)) +
+		"@" + fmt.Sprintf("%04X", 2500) + // royalties 25%
+		"@" + // hash
+		"@" + // attributes
+		"@" // uri
+}
+
+func registerTokens(
+	t *testing.T,
+	cs chainSim.ChainSimulator,
+	wallet dtos.WalletAddress,
+	nonce *uint64,
+	esdtSafeAddress []byte,
+	token chainSim.ArgsDepositToken,
+) {
+	ticker := getTokenTicker(token.Identifier)
+	registerCost, _ := big.NewInt(0).SetString(issuePaymentCost, 10)
+
+	registerData := "registerToken" +
+		"@" + hex.EncodeToString([]byte(token.Identifier)) +
+		"@" + fmt.Sprintf("%02x", uint32(token.Type)) +
+		"@" + hex.EncodeToString([]byte(ticker)) + // name
+		"@" + hex.EncodeToString([]byte(ticker)) + // ticker
+		"@" + hex.EncodeToString(big.NewInt(18).Bytes()) // num decimals
+	txResult := chainSim.SendTransaction(t, cs, wallet.Bytes, nonce, esdtSafeAddress, registerCost, registerData, uint64(100000000))
+	chainSim.RequireSuccessfulTransaction(t, txResult)
+
+	// wait for issue processing from metachain
+	err := cs.GenerateBlocks(3)
+	require.Nil(t, err)
+}
+
+func getTokenTicker(tokenIdentifier string) string {
+	return strings.Split(tokenIdentifier, "-")[1]
+}
+
+func generateHelloInAllShards(t *testing.T, cs chainSim.ChainSimulator) map[uint32]dtos.WalletAddress {
```

Comment:
Please change this function name.

Reply:
changed to `deployReceiverContractInAllShards`